rclpy/src/rclpy/_rclpy.c

```diff
@@ -879,6 +880,99 @@ rclpy_count_subscribers(PyObject * Py_UNUSED(self), PyObject * args)
   return _count_subscribers_publishers(args, "subscribers", rcl_count_subscribers);
 }
 
+typedef rcl_ret_t (* rcl_get_info_by_topic_func_t)(
+  const rcl_node_t * node,
+  rcutils_allocator_t * allocator,
+  const char * topic_name,
+  bool no_mangle,
+  rmw_topic_endpoint_info_array_t * info_array);
+
+static PyObject *
+_get_info_by_topic(
+  PyObject * args, const char * type,
+  rcl_get_info_by_topic_func_t rcl_get_info_by_topic)
+{
+  PyObject * pynode;
+  const char * topic_name;
+  int no_mangle;
+
+  if (!PyArg_ParseTuple(args, "Osp", &pynode, &topic_name, &no_mangle)) {
+    return NULL;
+  }
+
+  rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, "rcl_node_t");
+  if (!node) {
+    return NULL;
+  }
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rmw_topic_endpoint_info_array_t info_array = rmw_get_zero_initialized_topic_endpoint_info_array();
+  rcl_ret_t ret = rcl_get_info_by_topic(node, &allocator, topic_name, no_mangle, &info_array);
+  rmw_ret_t fini_ret;
+  if (ret != RCL_RET_OK) {
+    if (ret == RCL_RET_BAD_ALLOC) {
+      PyErr_Format(PyExc_MemoryError, "Failed to get information by topic for %s: %s",
+        type, rcl_get_error_string().str);
+    } else if (ret == RCL_RET_UNSUPPORTED) {
+      PyErr_Format(PyExc_NotImplementedError, "Failed to get information by topic for %s: "
+        "function not supported by RMW_IMPLEMENTATION", type);
+    } else {
+      PyErr_Format(RCLError, "Failed to get information by topic for %s: %s",
+        type, rcl_get_error_string().str);
+    }
+    rcl_reset_error();
+    fini_ret = rmw_topic_endpoint_info_array_fini(&info_array, &allocator);
+    if (fini_ret != RMW_RET_OK) {
+      PyErr_Format(RCLError, "rmw_topic_endpoint_info_array_fini failed: %s",
+        rmw_get_error_string().str);
+      rmw_reset_error();
+    }
+    return NULL;
+  }
+  PyObject * py_info_array = rclpy_convert_to_py_topic_endpoint_info_list(&info_array);
+  fini_ret = rmw_topic_endpoint_info_array_fini(&info_array, &allocator);
+  if (fini_ret != RMW_RET_OK) {
+    PyErr_Format(RCLError, "rmw_topic_endpoint_info_array_fini failed.");
+    rmw_reset_error();
+    return NULL;
+  }
+  return py_info_array;
+}
+
+/// Returns a list of publishers, publishing to a topic
```

Comment:
nitpick: `Return a list...` and also consider using punctuation

Reply:
Done.